src/particles/elements/RFCavity.H

```diff
@@ -0,0 +1,391 @@
+                   if( j % 2 == 0)
+                      scale_in = bgi;
+                   else
+                      scale_in = 1.0_prt;
+                   refpart.map(i,j) = refpart.map(i,j)*scale_in/scale_fin;
+               }
+            }
+
+            // advance integrated path length
+            refpart.s = s + slice_ds;
+        }
+
+        std::tuple<amrex::ParticleReal,amrex::ParticleReal,amrex::ParticleReal>
+        RF_Efield (amrex::ParticleReal const zeval) const
+        {
+            using namespace amrex::literals; // for _rt and _prt
+
+            // specify constants
+            amrex::ParticleReal const pi = 3.141592653589793_prt;
+            amrex::ParticleReal const zlen = 1.31879807_prt;
+            amrex::ParticleReal const zmid = zlen/2.0_prt;
+            int const ncoef = 25;
+
+            // specify Fourier coefficients
+            amrex::ParticleReal cos_coef [ncoef] = {};
+            amrex::ParticleReal sin_coef [ncoef] = {};
+
+            // recall array indexing begins with 0
+            cos_coef[0] = 0.1644024074311037;
+            cos_coef[1] = -0.1324009958969339;
+            cos_coef[2] = 4.3443060026047219E-002;
+            cos_coef[3] = 8.5602654094946495E-002;
+            cos_coef[4] = -0.2433578169042885;
+            cos_coef[5] = 0.5297150596779437;
+            cos_coef[6] = 0.7164884680963959;
+            cos_coef[7] = -5.2579522442877296E-003;
+            cos_coef[8] = -5.5025369142193678E-002;
+            cos_coef[9] = 4.6845673335028933E-002;
+            cos_coef[10] = -2.3279346335638568E-002;
+            cos_coef[11] = 4.0800777539657775E-003;
+            cos_coef[12] = 4.1378326533752169E-003;
+            cos_coef[13] = -2.5040533340490805E-003;
+            cos_coef[14] = -4.0654981400000964E-003;
+            cos_coef[15] = 9.6630592067498289E-003;
+            cos_coef[16] = -8.5275895985990214E-003;
+            cos_coef[17] = -5.8078747006425020E-002;
+            cos_coef[18] = -2.4044337836660403E-002;
+            cos_coef[19] = 1.0968240064697212E-002;
+            cos_coef[20] = -3.4461179858301418E-003;
+            cos_coef[21] = -8.1201564869443749E-004;
+            cos_coef[22] = 2.1438992904959380E-003;
+            cos_coef[23] = -1.4997753525697276E-003;
+            cos_coef[24] = 1.8685171825676386E-004;
+
+            // compute on-axis electric field (z is relative to cavity midpoint)
+            amrex::ParticleReal efield = 0.0;
+            amrex::ParticleReal efieldp = 0.0;
+            amrex::ParticleReal efieldpp = 0.0;
+            amrex::ParticleReal efieldint = 0.0;
+            amrex::ParticleReal const z = zeval - zmid;
+
+            if (abs(z)<=zmid)
+            {
+               efield = 0.5_prt*cos_coef[0];
+               efieldint = z*efield;
+               for(int j=1; j < ncoef; ++j)
+               {
+                 efield = efield + cos_coef[j]*cos(j*2*pi*z/zlen) +
+                      sin_coef[j]*sin(j*2*pi*z/zlen);
```

Comment:
I think `sin_coef` is still uninitialized at this point

Reply:
Doesn't line 199:

amrex::ParticleReal sin_coef [ncoef] = {};

initialize all elements of this array to 0.0_prt?